queries.sql

```diff
@@ -93,4 +93,67 @@ WHERE owners.full_name = 'Dean Winchester'
 AND animals.escape_attempts = 0;
 
 -- The one who owns the most animals
-SELECT owners.full_name, COUNT(animals.owner_id) FROM owners LEFT JOIN animals ON owners.id = animals.owner_id GROUP BY owners.full_name ORDER BY COUNT(animals.owner_id) DESC LIMIT 1;
\ No newline at end of file
+SELECT owners.full_name, COUNT(animals.owner_id) 
+FROM owners LEFT JOIN animals ON owners.id = animals.owner_id 
+GROUP BY owners.full_name ORDER BY COUNT(animals.owner_id) 
+DESC LIMIT 1;
+
+-- last animal seen by william
+SELECT animals.name, visits.date_of_visit FROM animals 
+JOIN visits ON animals.id = visits.animals_id 
+JOIN vets ON vets.id = visits.vet_id 
+WHERE vets.name = 'William Tatcher' 
+ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+-- The animals that stephanie saw
+SELECT vets.name, COUNT(*) FROM vets 
+JOIN visits ON visits.vet_id = vets.id 
+JOIN animals ON animals.id = visits.animals_id 
+WHERE vets.name = 'Stephanie Mendez' GROUP BY vets.name;
+
+-- All vets and specialties including those without specialties
+SELECT vets.name, species.name FROM vets 
+LEFT JOIN specializations ON vets.id = specializations.vet_id 
+LEFT JOIN species ON species.id = specializations.species_id;
+
+-- Animals that visited Stephanie Mendez between April 1st and August 30th, 2020
+SELECT animals.name, vets.name, visits.date_of_visit 
+FROM animals JOIN visits ON animals.id = visits.animals_id 
+JOIN vets ON vets.id = visits.vet_id 
+WHERE vets.name = 'Stephanie Mendez' 
+AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-31';
+
+-- Animal with most visit to vets
+SELECT animals.name, COUNT(*) FROM animals 
+JOIN visits ON animals.id = visits.animals_id 
+GROUP BY animals.name ORDER BY COUNT(*) DESC LIMIT 1;
+
+-- Vet maisy's first visit
+SELECT animals.name, vets.name, visits.date_of_visit 
+FROM animals JOIN visits 
+ON visits.animals_id = animals.id JOIN vets 
+ON vets.id = visits.vet_id WHERE vets.name = 'Maisy Smith' 
+ORDER BY visits.date_of_visit ASC LIMIT 1;
+
+-- Most recent visit: animal information, vet information, and date of visit
+SELECT * FROM animals JOIN visits 
+ON animals.id = visits.animals_id JOIN vets 
+ON vets.id = visits.vet_id 
+ORDER BY visits.date_of_visit DESC;
```

Comment:
- [ ] Ouch! this is a close one, but the query here should return a single record with details for the most recent visit instead you're returning multiple records about visits so please make sure to fix this.